core/src/main/java/org/apache/iceberg/hadoop/JdbcTableDao.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.hadoop;
+
+import java.sql.SQLException;
+import java.util.List;
+import org.apache.commons.dbutils.QueryRunner;
+import org.apache.commons.dbutils.handlers.BeanHandler;
+import org.apache.commons.dbutils.handlers.BeanListHandler;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class JdbcTableDao {
+  public static final String TABLES_TABLE_NAME = "iceberg_tables";
+  public static final String JDBC_CATALOG_TABLE_DDL =
+          "CREATE TABLE IF NOT EXISTS " + JdbcTableDao.TABLES_TABLE_NAME +
+                  "(catalogName VARCHAR(1255) NOT NULL," +
+                  "tableNamespace VARCHAR(1255) NOT NULL," +
+                  "tableName VARCHAR(1255) NOT NULL," +
+                  "metadataLocation VARCHAR(32768)," +
+                  "previousMetadataLocation VARCHAR(32768)," +
+                  "PRIMARY KEY (catalogName,tableNamespace,tableName)," +
+                  "FOREIGN KEY (catalogName, tableNamespace) REFERENCES " + JdbcNamespaceDao.NAMESPACES_TABLE_NAME +
+                  "(catalogName, namespace)" +
+                  ");";
+  private static final Logger LOG = LoggerFactory.getLogger(JdbcTableDao.class);
+  private final QueryRunner queryRunner;
+  private final String catalogName;
+
+
+  public JdbcTableDao(QueryRunner queryRunner, String catalogName) {
+    this.queryRunner = queryRunner;
+    this.catalogName = catalogName;
+  }
+
+  public boolean isExists(TableIdentifier tableIdentifier) {
+    try {
+      return this.get(tableIdentifier) != null;
+    } catch (SQLException throwables) {
```

Comment:
Any SQL exception? What possible exceptions does this include?

Reply:
i would expect potential exceptions like 'when the database is down', 'not reachable' or 'table is not there'. otherwise the result shuld return = null 
maybe its better to re-throw the exception.. otherwise catalog class will assume table is not exists and will continue.